src/workloads/query/CollectionLevelDiagnosticCommands.yml

```diff
@@ -0,0 +1,176 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query-execution"
+Description: |
+  This workload measures performance for diagnostic top command.
+
+Keywords:
+- top
+- command
+- admin
+- stats
+
+GlobalDefaults:
+  Database: &Database test
+  CollectionCount: &CollectionCount 200
+  DocumentCount: &DocumentCount 1e5
+  MaxPhases: &MaxPhases 10
+
+Clients:
+  Default:
+    QueryOptions:
+      maxPoolSize: 2000
+
+Actors:
+# Clear any pre-existing collection state.
+- Name: ClearCollection
+  Type: CrudActor
+  CollectionCount: *CollectionCount
+  Database: *Database
+  Threads: *CollectionCount
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: *CollectionCount
+        CollectionCount: *CollectionCount
+        Operations:
+        - OperationName: drop
+
+- Name: InsertData
+  Type: Loader
+  Threads: *CollectionCount
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        MultipleThreadsPerCollection: true
+        CollectionCount: *CollectionCount
+        DocumentCount: *DocumentCount
+        BatchSize: 1000
+        Document:
+          int: &integer {^Choose: {from: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10], deterministic: true}}
+          string: &string {^Choose: {from: ["swimswimswimswimswim", "sinksinksinksink"], deterministic: true}}
+          obj: {int: *integer, string: *string}
+
+# Phase 2: Ensure all data is synced to disk.
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [2]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+
+- Name: DiagnosticTopCommand
+  Type: RunCommand
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [3, 5, 7, 9]
```

Comment:
Can you also run the top with the below commands concurrently?

Reply:
This would be a good addition. I added a new actor "DiagnosticConcurrentTopCommand" to run in parallel with the CRUD operations.